### PR TITLE
Docs: Fix build 3 warnings

### PR DESCRIPTION
**Pull request by benjaoming**
2 warnings were caused by extensions that didn't declare to be parallel safe.
1 warning was do to a parallel bug in sphinxcontrib-httpdomain that got fixed in 1.8.1.

I also updated the Makefile to run a build that looks more like the Read the Docs build, so the chances to discover these issues locally are a bit higher.

### CODE REVIEW

File: requirements/docs.in [hunk 1]
```diff
@@ -3,18 +3,22 @@
 -r pip.txt
 
 sphinx_rtd_theme==1.1.0
+# Note: Version 3.4.1 of sphinx-tabs requires docutils 0.18 which is yet to be supported by sphinx-rtd-theme
+# Version 3.4.0 has an incompatible Jinja2 version which also blocks sphinx-rtd-theme
+# All-together, we cannot upgrade to Sphinx 5.x before either sphinx-tabs or sphinx-rtd-theme fixes these
+# issues.
 sphinx-tabs==3.3.1
```
Comment (humitos) [inline requirements/docs.in:10]: We can probably migrate `sphinx-tabs` to `sphinx-design`: https://sphinx-design.readthedocs.io/en/latest/tabs.html at some point

**Review by humitos — APPROVED** [review]
This is good! 👍🏼 

The next obvious step here is to unpin all the packages from `requirements/docs.in`; keeping pinned only the ones that are required to be pinned. This way, our GitHub action auto-updates them weekly following the same pattern we have with the rest of the requirements and we don't have to care about these type of PRs

File: requirements/docs.in (hunk 1 above)
Comment (benjaoming) [reply requirements/docs.in:10]: That would be really great. 1) to have one extension less and 2) I got an impression that sphinx-tabs is less maintained nowadays.

**Comment by benjaoming** [comment]
I was somehow under the impression that the dev docs were building fine. But they have some broken intersphinx references. I'll fix that as well...

> The next obvious step here is to unpin all the packages from requirements/docs.in;

YES :100:

**Comment by benjaoming** [comment]
> I was somehow under the impression that the dev docs were building fine. But they have some broken intersphinx references. I'll fix that as well...

These 503 errors were temporary. Has there been a known disturbance or should this be escalated separately?

```
WARNING: failed to reach any of the inventories with the following issues:
intersphinx inventory 'https://myst-parser.readthedocs.io/en/stable/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Unavailable for url: https://myst-parser.readthedocs.io/en/stable/objects.inv
loading intersphinx inventory from https://docs.readthedocs.io/en/stable/objects.inv...
WARNING: failed to reach any of the inventories with the following issues:
intersphinx inventory 'https://jupyterbook.org/en/stable/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: 503 Server Error: Service Unavailable for url: https://jupyterbook.org/en/stable/objects.inv
loading intersphinx inventory from https://dev.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://docs.jupyter.org/en/latest/objects.inv...
```

**Comment by humitos** [comment]
> These 503 errors were temporary. Has there been a known disturbance or should this be escalated separately?

Escalated to where? Those 503 errors on those URLs are from our own service.

**Pull request merged**